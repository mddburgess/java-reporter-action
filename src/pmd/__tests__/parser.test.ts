import * as glob from "@actions/glob";
import PmdParser from "../parser";

describe("PmdParser", () => {
  it("can parse an empty PMD report", () => {
    const report = new PmdParser("src/pmd/__tests__/__fixtures__/empty.xml").read();
    expect(report).toMatchSnapshot();
  });

  it("can parse a real PMD report", async () => {
    const globber = await glob.create("**/target/pmd.xml");
    const paths = await globber.glob();
    expect(paths.length).toBeGreaterThan(0);

    const report = new PmdParser(paths[0]).read();
    expect(report).toMatchSnapshot();
  });
});
