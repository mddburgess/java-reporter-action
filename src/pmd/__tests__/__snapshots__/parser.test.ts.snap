// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PmdParser can parse a real PMD report 1`] = `
Object {
  "violations": Array [
    Object {
      "endColumn": 1,
      "endLine": 5,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/pmd/bestpractices/AbstractClassWithoutAbstractMethod.java",
      "message": "Each class should declare at least one constructor.",
      "priority": "3",
      "rule": "AtLeastOneConstructor",
      "ruleset": "Code Style",
      "startColumn": 17,
      "startLine": 3,
    },
    Object {
      "endColumn": 1,
      "endLine": 5,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/pmd/bestpractices/AbstractClassWithoutAbstractMethod.java",
      "message": "No abstract method which means that the keyword is most likely used to prevent instantiation. Use a private or protected constructor instead.",
      "priority": "1",
      "rule": "AbstractClassWithoutAnyMethod",
      "ruleset": "Design",
      "startColumn": 17,
      "startLine": 3,
    },
    Object {
      "endColumn": 1,
      "endLine": 5,
      "filePath": "/Users/mike/Projects/java-reporter-action/example/src/main/java/pmd/bestpractices/AbstractClassWithoutAbstractMethod.java",
      "message": "This abstract class does not have any abstract methods.",
      "priority": "3",
      "rule": "AbstractClassWithoutAbstractMethod",
      "ruleset": "Best Practices",
      "startColumn": 17,
      "startLine": 3,
    },
  ],
}
`;

exports[`PmdParser can parse an empty PMD report 1`] = `
Object {
  "violations": Array [],
}
`;
